File: 0041-first-missing-positive/0041-first-missing-positive.py
class Solution:
    def firstMissingPositive(self, nums: List[int]) -> int:
        n = len(nums)+2
        dp = [1]*n
        dp[0] = 0
        for num in nums:
            if n > num >= 1:
                dp[num] = 0

        return dp.index(1)

File: 0041-first-missing-positive/0041-first-missing-positive.cpp
class Solution {
public:
    int firstMissingPositive(vector<int>& nums) {
        int mx = *max_element(nums.begin(), nums.end());
        if(mx<0) return 1;
        unordered_map<int, int>mp;
        for(int x  : nums)mp[x]++;
        for(int i=1; i<=mx; i++){
            if(!mp.count(i)) return i;
        }
        return mx+1;
    }
};